build-logic/integration-testing/src/main/kotlin/gradlebuild.cross-version-tests.gradle.kts

```diff
@@ -51,9 +52,11 @@ fun createQuickFeedbackTasks() {
         val testTask = createTestTask(taskName, executer, sourceSet, testType) {
             this.systemProperties["org.gradle.integtest.versions"] = "latest"
             this.systemProperties["org.gradle.integtest.crossVersion"] = "true"
-            this.useJUnitPlatform {
-                includeEngines("cross-version-test-engine")
-            }
+
+            // We should always be using JUnitPlatform at this point, so don't call useJUnitPlatform(), else this will
+            // discard existing options configuration and add a deprecation warning.  Just set the existing options.
+            assert(JUnitPlatformTestFramework::class.java == this.testFramework.javaClass)
```

Comment:
Do we need the assert here? I suppose the next line would fail with a clast cast exception anyway.